### ISSUE TITLE
"Connection to server is erroring. Shutting down server."

### ISSUE THREAD

**Issue by Spongman** [microsoft/vscode-cpptools#276]
- VSCode Version: code_1.5.3-1474534134_i386.deb
- OS Version: Linux 3.16.0-4-686-pae #1 SMP Debian 3.16.36-1+deb8u1 (2016-09-03) i686 GNU/Linux

Steps to Reproduce:
1. install vscode-cpptools
2. open output window

output window contains 3x "Connection to server is erroring. Shutting down server."

broken: autocomplete, symbol search, goto/peek definition.

f12 shows script errors:

```
workbench.main.js:118
[Extension Host] TypeError: Cannot read property 'forEach' of null
    at LanguageClient.cleanUp (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/lib/main.js:471:24)
    at LanguageClient.stop (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/lib/main.js:454:14)
    at LanguageClient.handleConnectionError (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/lib/main.js:684:18)
    at errorHandler (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/lib/main.js:559:19)
    at /home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/lib/main.js:45:42
    at CallbackList.invoke (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/events.js:103:39)
    at Emitter.fire (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/events.js:167:36)
    at writeErrorHandler (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:113:22)
    at CallbackList.invoke (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/events.js:103:39)
    at Emitter.fire (/home/piersh/.vscode/extensions/ms-vscode.cpptools-0.9.2/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/events.js:167:36)(anonymous function) @ workbench.main.js:118
```

**Comment by Spongman** [comment]
seriously? where's the 32-bit binary?

**Comment by sean-mcmanus** [comment]
It's not built. We also don't support all Linux distributions (e.g. Ubuntu 12, CentOS7...I don't know which 64-bit Debian versions we work on). Seems like we could show a better error message at least.

**Comment by Spongman** [comment]
can you build the 32-bit version?

**Comment by sean-mcmanus** [comment]
Unfortunately, we're not currently planning to, and it's not as trivial as changing a compiler flag. We also don't have easy access to 32-bit Linux machines (Azure VMs don't support 32-bit OS's). We expect most Linux users to be using a 64-bit version, so I would recommend upgrading to a 64-bit OS if possible or using another 64-bit Linux OS. Is there are reason you're using a 32-bit OS? Maybe this could change if we get enough feedback or telemetry. You could also try using other VS code C++ extensions that supports 32-bit.

**Comment by bobbrow** [comment]
Consolidating the 32-bit Linux issues...  We are currently tracking this issue with #424.  Thank you.

**Comment by sean-mcmanus** [comment]
Fixed with 0.14.4.
